kafka-rest/src/test/java/io/confluent/kafkarest/controllers/BrokerConfigManagerImplTest.java

```diff
@@ -0,0 +1,267 @@
+      new BrokerConfig(
+          CLUSTER_ID,
+          BROKER_ID,
+          "config-3",
+          "value-3",
+          /* isDefault= */false,
+          /* isReadOnly= */false,
+          /* isSensitive= */true);
+
+  private static final Config CONFIG =
+      new Config(
+          Arrays.asList(
+              new ConfigEntry(
+                  CONFIG_1.getName(),
+                  CONFIG_1.getValue(),
+                  CONFIG_1.isDefault(),
+                  CONFIG_1.isSensitive(),
+                  CONFIG_1.isReadOnly()),
+              new ConfigEntry(
+                  CONFIG_2.getName(),
+                  CONFIG_2.getValue(),
+                  CONFIG_2.isDefault(),
+                  CONFIG_2.isSensitive(),
+                  CONFIG_2.isReadOnly()),
+              new ConfigEntry(
+                  CONFIG_3.getName(),
+                  CONFIG_3.getValue(),
+                  CONFIG_3.isDefault(),
+                  CONFIG_3.isSensitive(),
+                  CONFIG_3.isReadOnly())));
+
+  @Rule
+  public final EasyMockRule mocks = new EasyMockRule(this);
+
+  @Mock
+  private Admin adminClient;
+
+  @Mock
+  private ClusterManager clusterManager;
+
+  @Mock
+  private DescribeConfigsResult describeConfigsResult;
+
+  @Mock
+  private AlterConfigsResult alterConfigsResult;
+
+  private BrokerConfigManagerImpl brokerConfigManager;
+
+  @Before
+  public void setUp() {
+    brokerConfigManager = new BrokerConfigManagerImpl(adminClient, clusterManager);
+  }
+
+  @Test
+  public void listBrokerConfigs_existingBroker_returnsConfigs() throws Exception {
+    expect(clusterManager.getCluster(CLUSTER_ID)).andReturn(completedFuture(Optional.of(CLUSTER)));
+    expect(
+        adminClient.describeConfigs(
+            singletonList(new ConfigResource(Type.BROKER, BROKER_ID))))
+            .andReturn(describeConfigsResult);
+    expect(describeConfigsResult.values())
+        .andReturn(
+            singletonMap(
+                new ConfigResource(Type.BROKER, BROKER_ID),
+                KafkaFuture.completedFuture(CONFIG)));
+    replay(adminClient, clusterManager, describeConfigsResult);
+
+    List<BrokerConfig> configs = brokerConfigManager.listBrokerConfigs(CLUSTER_ID, BROKER_ID).get();
+
+    assertEquals(new HashSet<>(Arrays.asList(CONFIG_1, CONFIG_2, CONFIG_3)),
+        new HashSet<>(configs));
+  }
+
+  @Test
+  public void listBrokerConfigs_nonExistingBroker_throwsNotFound() throws Exception {
+    expect(clusterManager.getCluster(CLUSTER_ID)).andReturn(completedFuture(Optional.of(CLUSTER)));
+    expect(
+        adminClient.describeConfigs(
+            singletonList(new ConfigResource(Type.BROKER, BROKER_ID))))
+        .andReturn(describeConfigsResult);
+    expect(describeConfigsResult.values())
+        .andReturn(
+            singletonMap(
+                new ConfigResource(Type.BROKER, BROKER_ID),
+                TestUtils.failedFuture(new BrokerNotAvailableException("Broker not found."))));
```

Comment:
Instead of the generic `NotFound` exception, thinking whether we can use `BrokerNotAvailableException` in case of invalid broker id. Would like to know your thoughts.

Reply:
It think we should ignore proper error reporting for now. We took a lot of shortcuts for 5.5 on that. I will re-do error handling for all v3 APIs later.